### PR TITLE
[16.0][ADD] shopinvader_api_cart: cart helper: Add hooks

### PR DESCRIPTION
**Pull request by mmequignon**
Add hooks in cart helper in order to ease override when modifying transaction schema

### CODE REVIEW

File: shopinvader_api_cart/routers/cart.py [hunk 1]
```diff
@@ -124,9 +124,22 @@ def _apply_transactions_on_existing_cart_line(
         ):
             return (2, cart_line.id, None)
         vals = {"product_uom_qty": new_qty}
+        vals = self._apply_transactions_on_existing_cart_line__post(
```
Comment (simahawk) [inline shopinvader_api_cart/routers/cart.py:127]:
```suggestion
        vals = self._apply_transactions_on_existing_cart_line_prepare_vals(
```

File: shopinvader_api_cart/routers/cart.py [hunk 2]
```diff
@@ -138,10 +151,23 @@ def _apply_transactions_creating_new_cart_line(
             cart=cart, transactions=transactions
         )
         if vals:
+            vals = self._apply_transactions_creating_new_cart_line__post(
```
Comment (simahawk) [inline shopinvader_api_cart/routers/cart.py:157]:
```suggestion
            vals = self._apply_transactions_creating_new_cart_line_prepare_vals(
```

**Review by marielejeune — APPROVED** [review]

**Comment by lmignon** [comment]
@mmequignon Can you take into account the comments from Simone plz?

**Comment by paradoxxxzero** [comment]
I've done the changes in #1571, we need to go forward

**Pull request closed without merging**